client/dashboard/sites/monitoring-request-methods-card/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { siteMetricsQuery } from '@automattic/api-queries';
+import { DataPointPercentage, PieChart, Legend } from '@automattic/charts';
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import MonitoringCard from '../monitoring-card';
+import type { LegendData, PeriodData, TimeRange } from '../monitoring/types';
+import type { Site, SiteHostingMetrics } from '@automattic/api-core';
+
+function convertTimeRangeToUnix( timeRange: number ): TimeRange {
+	const start = Math.floor( new Date().getTime() / 1000 ) - timeRange * 3600;
+	const end = Math.floor( new Date().getTime() / 1000 );
+
+	return { start, end };
+}
+
+type SiteRequestMethodsData = {
+	data: DataPointPercentage[];
+	isLoading: boolean;
+};
+
+const chartColors = [ '#3858E9', '#5BA300', '#F57600', '#B51963' ];
+
+function useSiteRequestMethodsData( siteId: number, timeRange: number ): SiteRequestMethodsData {
+	const { start, end } = useMemo( () => convertTimeRangeToUnix( timeRange ), [ timeRange ] );
+
+	const { data: requestMethodsData, isPending: isLoading } = useQuery(
+		siteMetricsQuery( siteId, { start, end, metric: 'requests_persec', dimension: 'http_verb' } )
+	);
+
+	const formatData = ( requestMethodsData: SiteHostingMetrics ): DataPointPercentage[] => {
+		if ( ! requestMethodsData?.data?.periods ) {
+			return [];
+		}
+
+		const methodsMap: {
+			[ key: string ]: number;
+		} = {};
+
+		requestMethodsData.data.periods.forEach( ( period: PeriodData ) => {
+			if ( typeof period?.dimension === 'object' ) {
+				Object.entries( period.dimension ).forEach( ( [ method, value ] ) => {
+					if ( ! methodsMap[ method ] ) {
+						methodsMap[ method ] = 0;
+					}
+					methodsMap[ method ] += value;
+				} );
+			}
+		} );
+
+		const sum: number = Object.values( methodsMap ).reduce(
+			( acc: number, curr: number ): number => acc + curr,
+			0
+		);
+		const chartColorsCopy = [ ...chartColors ];
+
+		return Object.entries( methodsMap ).map(
+			( [ method, value ]: [ string, number ] ): DataPointPercentage => ( {
+				label: method.toUpperCase(),
+				value: Math.round( value * 100 ) / 100,
+				percentage: Math.round( ( value * 100 ) / sum ),
+				valueDisplay: Math.round( ( value * 100 ) / sum ).toString() + '%',
+				color: chartColorsCopy.shift() || '#000000',
```

Comment:
Nit: Can we use index here from `map` and just iterate through chartColorsCopy? Something like:

`chartColors[ index % chartColors.length ]`

Reply:
Good point, done: https://github.com/Automattic/wp-calypso/pull/105863/commits/2b565808ae4ea6c5b54508a1fb0e34c83ffa3dcb